python/pyspark/serializers.py

```diff
@@ -298,7 +298,10 @@ def create_array(s, t):
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             try:
-                array = pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)
+                if str(s.dtype) == 'category':
+                    array = pa.array(s.get_values())
```

Comment:
Can we have a way to convert from/to PySpark and Pandas in roundtrip?

Reply:
Hello, 

I am newbie in python.  is this correct way to convert from/to PySpark and pandas? 
```
array = pa.Array.from_pandas(s.get_values(), mask=mask, type=t, safe=self._safecheck)
```

Reply:
Hm, I am not sure how we will be able to convert Spark DataFrame -> pandas DataFrame with a category type. 

pandas DataFrame -> Spark DataFrame looks easy as you did.

Reply:
I can think of introducing arrow dictionary type in Spark DataFrame, this way we can convert Spark DataFrame -> pandas DataFrame -> Spark DataFrame. 

However, this will change DataFrame types in Arrow mode vs non Arrow mode. 
For example, 
```
>>> spark.conf.set("spark.sql.execution.arrow.enabled", False)
>>> df = spark.createDataFrame(pdf)
>>> df.printSchema()
root
 |-- A: string (nullable = true)
 |-- B: string (nullable = true)

-------
>>> spark.conf.set("spark.sql.execution.arrow.enabled", True)
>>> df = spark.createDataFrame(pdf)
>>> df.printSchema()
root
 |-- A: string (nullable = true)
 |-- B: dictionary 
           - values: string (nullable = true)

```
I am not sure if this may break some use case or correct way to do it.

Reply:
@BryanCutler

Hi bryan, 
I need help here, since you identified this, looping you in. 
this change has been blocked for a long time
can you provide some insight/guidance here?

Reply:
The JIRA was meant to be a 1 way conversion from pandas to pyspark as the decoded type.  Making a similar categorical type in pyspark might be possible, but that's out of scope of the intent here.

Reply:
understood, will cut a new JIRA for this